proc_macros/src/lib.rs

```diff
@@ -165,3 +203,599 @@ pub fn rule_tests(input: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+enum RuleStateScope {
+    RuleStatic,
+    PerRun,
+    PerFileRun,
+}
+
+impl Parse for RuleStateScope {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let content;
+        bracketed!(content in input);
+        let found = match &*content.parse::<Ident>()?.to_string() {
```

Comment:
There's got to be a better way to parse against a couple different literal sequences of characters with `syn` (here I'm just trying to match `[rule-static]`, `[per-run]` or `[per-file-run]`)?

Reply:
Wait, how does this work? The identifier is just the "rule" part because identifiers cant include `-`? Is it weird that you're using `-` in things that look identifier-ish?

Reply:
Right since a "kebab-case identifier" like `per-run` isn't a "thing" in Rust I guess there aren't like "primitives" for parsing it in `syn`?

But what it seems like there should be a primitive for is "just parse this exact sequence of characters"?

> Is it weird

I think I like it "aesthetically" here, reminds of eg "section headers" or whatever in TOML files